### ISSUE TITLE
Minified/customisable version of the PSL

### ISSUE THREAD

**Issue by wdhdev** [publicsuffix/list#2256]
Would it be worth offering a minified version of the PSL? It would be good for companies that use the PSL but do not need any comments or anything and just want the rules.

So basically just offering a version of the PSL without any comments.

**Comment by weppos** [comment]
> Would it be worth offering a minified version of the PSL? It would be good for companies that use the PSL but do not need any comments or anything and just want the rules.
> 
> So basically just offering a version of the PSL without any comments.

It's one extra source to maintain, and I can't see a lot of value to produce it on our side. It's very easy to transform the list in whatever format you want.

Moreover, different languages/applications may use a different approach for minification or optimization. In many cases, an extra transformation is still needed to compile the list into something more optimized (many consumers concerned about performance have that intermediate step).

Adding one extra minified version will add extra maintenance overhead for very little gain.

My 2 cents.

**Comment by wdhdev** [comment]
> It's one extra source to maintain, and I can't see a lot of value to produce it on our side. It's very easy to transform the list in whatever format you want.

That is true, however I believe if it was implemented it would receive quite a bit of use as most people are not using, nor relying on the comments in the file. [The comments in the file increase the file size by over 55%](https://github.com/wdhdev/psl-min/commit/7206290c011e2b51b2d4ef833aea0594f598d130).

> Moreover, different languages/applications may use a different approach for minification or optimization. In many cases, an extra transformation is still needed to compile the list into something more optimized (many consumers concerned about performance have that intermediate step).

This is true, however for some basic level consumers who are using a mostly unmodified version of the PSL, it could be helpful.

A bit unrelated, it could be worth deploying separate ICANN and Private section only files, which I know would benefit some consumers, as I know some only want ICANN section TLDs, while others just want Private section domains.

**Comment by mozfreddyb** [comment]
To be honest, I would actually prefer we go into another direction and move to a better, more expressive format that goes beyond comments where we have better metadata in the entries as dictionaries (json maybe?) so that we can extend the flags in the future. We could still provide the `dat` as a compile-artifact, but eventually I can see us opening this up the other way around.

**Comment by wdhdev** [comment]
If we were going to use JSON, it might be a good idea to have individual JSON files for each individual company listed on the PSL (at least in the repo for easy editing) but then combining them all into one huge file on the web server.

**Comment by weppos** [comment]
> To be honest, I would actually prefer we go into another direction and move to a better, more expressive format that goes beyond comments where we have better metadata in the entries as dictionaries (json maybe?) so that we can extend the flags in the future. We could still provide the `dat` as a compile-artifact, but eventually I can see us opening this up the other way around.

100% agreed. If we have to add an additional format/version, I'd rather use something more expressive and more conveniently parsable.

**Comment by wdhdev** [comment]
JSON would definitely be a better option, it's much easier to parse as you say.

**Comment by wdhdev** [comment]
A lot of what you've said there could definitely be doable with little resourcing, simply even could be done via a small API.

e.g. A request to https://publicsuffix.org/list/public_suffix_list.dat?exclude_comments=true would return a version of the PSL without comments. This sort of basic formatting could be easily done via a static web page with client side JS.
